### ISSUE TITLE
A detail in SIZE(/struct) for HASH() and LIST()

### ISSUE THREAD

**Issue by alaingdl** [gnudatalanguage/gdl#1807]
Working in cleaning some other tests, I got this _small_ mistake for HASH() and LIST().

```
h = HASH('a',1,'b',2,'c',3)
strucobj = SIZE(h,/struct)
print,(strucobj).dimensions

GDL return : 
           0           0           0           0           0           0           0           0
IDL return :
           3           0           0           0           0           0           0           0
(8 dimensions, 3 elements in the first one)
```

and for LIST()
```
l = LIST('a',1,'b',2,'c',3)
strucobj = SIZE(l,/struct)
print,(strucobj).dimensions

GDL return : 
           0           0           0           0           0           0           0           0
IDL return :
           6           0           0           0           0           0           0           0
(8 dimensions, 6 elements in the first one)
```

The value is OK for arrays (1, [1], [1,2], ...)

**Comment by alaingdl** [comment]
just to confirm that this problem don't exist for true structures, e.g. (as in `test_structures.pro`)

```
 structarray=REPLICATE({test, value:0.0},10)
 size(STRUCTARRAY,/stru)
 size(STRUCTARRAY[0],/stru)
```
same outputs in IDL & GDL :)

(but I will try to add that somewhere in the test suite)

**Comment by alaingdl** [comment]
ok, also `print,(strucobj).n_dimensions` is wrong in both cases (but OK for structures ...)

**Comment by alaingdl** [comment]
These bugs are corrected in #1815 

Please open a new issue if related problems found, with related snippets.
